R/utility.R

```diff
@@ -725,4 +724,117 @@ viewSplines <- function(knots, degree, theta) {
   }
 
   return(p)
-}
\ No newline at end of file
+}
+
+#' Get survival curve parameters
+#'
+#' @param points A list of two-element vectors specifying the desired time and 
+#' probability pairs that define the desired survival curve
+#' @return A vector of parameters that define the survival curve optimized for
+#' the target points. The first element of the vector represents the "f"
+#' parameter and the second element represents the "shape" parameter.
+#' @examples
+#' points <- list(c(60, 0.90), c(100, .75), c(200, .25), c(250, .10))
+#' survGetParams(points)
+#' @export
+#' @concept utility
+survGetParams <- function(points) {
+  
+  assertNotMissing(points = missing(points))
+  assertClass(points = points, class = "list")
+  
+  time <- sapply(points, function(x) x[1])
```

Comment:
As described [here](https://adv-r.hadley.nz/functionals.html?q=sapply#purrr-shortcuts) (scroll up a bit from the linked section) it is safer to use vapply then sapply. `FUN.VALUE` describes a template of the return value of the function, here simply a numeric of length 1. and will error on anything else. sapply might try other simplifications that could cause issues. 
```suggestion
  time <- vapply(points, function(x) x[1], FUN.VALUE = numeric(1))
```